dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -126,50 +126,22 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
                 m_registering = false;
             }) };
 
-            winrt::guid storedComActivatorGuid{ GUID_NULL };
-            if (!PushNotificationHelpers::IsPackagedAppScenario())
+            winrt::guid registeredClsid{};
+            if (AppModel::Identity::IsPackagedProcess())
             {
-                if (!AppModel::Identity::IsPackagedProcess())
-                {
-                    THROW_IF_FAILED(PushNotifications_RegisterFullTrustApplication(m_appId.c_str(), GUID_NULL));
-
-                    storedComActivatorGuid = RegisterComActivatorGuidAndAssets();
-                }
-
-                if (!WindowsAppRuntime::SelfContained::IsSelfContained())
-                {
-                    auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
-                    THROW_IF_FAILED(notificationPlatform->AddToastRegistrationMapping(m_processName.c_str(), m_appId.c_str()));
-                }
+                registeredClsid = RegisterPackagedApp();
             }
-
-            winrt::guid registeredClsid{ GUID_NULL };
-            if (AppModel::Identity::IsPackagedProcess())
+            else
             {
-                registeredClsid = PushNotificationHelpers::GetComRegistrationFromRegistry(expectedAppServerArgs.data());
+                AppNotificationAssets assets{ GetAssets() };
```

Comment:
Can we use std::move semantics to avoid excessive copies here and below?

Reply:
Actually, we should probably leave this to the compiler. Based on what I am reading here: https://en.cppreference.com/w/cpp/language/copy_elision - the compiler tries to do a return value optimization first i.e.-"objects are constructed directly into the storage where they would otherwise be copied/moved to." - if it fails then it would switch to move and then finally to copy. Trying to std::move would end up counterintuitive to what is being achieved. I would be avoiding this in the PR. Let me know if you otherwise. Thanks.